src/modules/tasks.js

```diff
@@ -0,0 +1,141 @@
+
+// Create URL from supplied path and options
+function createTaskUrl(path, options, additionalQueryParams = {}, apiVersion = 'v1') {
+  const {
+    apiKey,
+    serviceUrl,
+  } = options;
+  let queryParams = {
+    ...additionalQueryParams,
+  };
+
+  // Validate path is provided
+  if (!path || typeof path !== 'string') {
+    throw new Error('path is a required parameter of type string');
+  }
+
+  queryParams.key = apiKey;
+  queryParams = helpers.cleanParams(queryParams);
+
+  const queryString = qs.stringify(queryParams, { indices: false });
+
+  return `${serviceUrl}/${encodeURIComponent(apiVersion)}/${encodeURIComponent(path)}?${queryString}`;
+}
+
+/**
+ * Interface to task related API calls
+ *
+ * @module tasks
+ * @inner
+ * @returns {object}
+ */
+class Tasks {
+  constructor(options) {
+    this.options = options || {};
+  }
+
+  /**
+   * Retrieve all tasks from index
+   *
+   * @function getAllTasks
+   * @param {object} parameters - Additional parameters for task details
+   * @param {number} [parameters.num_results_per_page] - The number of tasks to return. Defaults to 20. Maximum value 100
+   * @param {number} [parameters.page] - The page of results to return. Defaults to 1
+   * @param {object} [networkParameters] - Parameters relevant to the network request
+   * @param {number} [networkParameters.timeout] - Request timeout (in milliseconds)
+   * @returns {Promise}
+   * @see https://docs.constructor.io/rest_api/tasks/#retrieve-all-tasks
+   */
+  getAllTasks(parameters = {}, networkParameters = {}) {
+    const queryParams = {};
+    let requestUrl;
+    const fetch = (this.options && this.options.fetch) || nodeFetch;
+    const controller = new AbortController();
+    const { signal } = controller;
+
+    if (parameters) {
+      const { num_results_per_page: numResultsPerPage, page } = parameters;
+
+      // Pull number of results per page from parameters
+      if (numResultsPerPage) {
+        queryParams.num_results_per_page = numResultsPerPage;
+      }
+
+      // Pull page from parameters
+      if (page) {
+        queryParams.page = page;
+      }
+    }
+
+    try {
+      requestUrl = createTaskUrl('tasks', this.options, queryParams);
+    } catch (e) {
+      return Promise.reject(e);
+    }
+
+    helpers.applyNetworkTimeout(this.options, networkParameters, controller);
+
+    return fetch(requestUrl, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        ...helpers.createAuthHeader(this.options),
+      },
+      signal,
+    }).then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+
+      return helpers.throwHttpErrorFromResponse(new Error(), response);
+    }).then((json) => json);
+  }
+
+  /**
+   * Retrieve task given a specific id
+   *
+   * @function getTask
+   * @param {object} parameters - Additional parameters for task details
+   * @param {string} parameters.id - The ID of the task you'd like to retrieve
```

Comment:
Generally try to avoid "you", "you'd", etc.

Reply:
Roger dodger, updated